public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx

```diff
@@ -222,6 +230,16 @@ export function RichHistoryQueriesTab(props: Props) {
             </div>
           );
         })}
+        <FullWidthButtonContainer>
+          <Button
+            variant="secondary"
+            onClick={() => {
+              setRenderLimit(renderLimit + 30);
+            }}
+          >
+            Show more queries{' '}
+          </Button>
```

Comment:
Do we still keep the query history limit on 100 queries? If so, should we render 30 more on each click then? Also, we should probably hide the button when we reach the end of the list.

Reply:
>Do we still keep the query history limit on 100 queries?

No, we don't have limit at the moment. We just delete queries that are older than x days( user can set this up for up 2 weeks)

>Also, we should probably hide the button when we reach the end of the list.

Yes, I absolutely agree. This is just a draft.  Still need a feedback from UX on how to set it up. :)